### ISSUE TITLE
UWP authentication crash ((WAB) authentication failed)

### ISSUE THREAD

**Issue by charlesroddie** [AzureAD/microsoft-authentication-library-for-dotnet#1782]
Error from an end-user system. (cc @kevcrooks)

Microsoft.Identity.Client 4.11.0
UWP, WINDOWS 10.0.18362
Desktop, Interactive authentication flow
Using Azure AD B2C
We have provided the internetClient capability.

Expected behaviour: the AcquireTokenInteractive box should appear after the ExecuteAsync() method is called, for the user to log in

Actual behaviour: The box does not appear, and an exception is thrown. Report from App Center:

```
Exception: Microsoft.Identity.Client.MsalClientException
Message: Web Authentication Broker (WAB) authentication failed. To collect WAB logs, please follow https://aka.ms/msal-net-wab-logs
StackTrace: 
Microsoft.Identity.Client.Platforms.uap.WebUI.<AcquireAuthorizationAsync>d__6.MoveNext() + 0x2cb
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x70
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17
   at Microsoft.Identity.Client.Internal.AuthCodeRequestComponent.<FetchAuthCodeAndPkceInternalAsync>d__6.MoveNext() + 0x202
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x70
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17
   at Microsoft.Identity.Client.Internal.AuthCodeRequestComponent.<FetchAuthCodeAndPkceVerifierAsync>d__4.MoveNext() + 0xed
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x70
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17
   at Microsoft.Identity.Client.Internal.Requests.InteractiveRequest.<GetTokenResponseAsync>d__11.MoveNext() + 0x374
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x70
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17
   at Microsoft.Identity.Client.Internal.Requests.InteractiveRequest.<ExecuteAsync>d__8.MoveNext() + 0x293
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x70
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17
   at Microsoft.Identity.Client.Internal.Requests.RequestBase.<RunAsync>d__14.MoveNext() + 0x488
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x70
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17
   at Microsoft.Identity.Client.ApiConfig.Executors.PublicClientExecutor.<ExecuteAsync>d__2.MoveNext() + 0x170
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x70
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17
   at FSharp.Control.Tasks.TaskBuilder.bindTask@121.Invoke(Unit) + 0x1b
   at FSharp.Control.Tasks.TaskBuilder.tryWith[a](FSharpFunc`2, FSharpFunc`2) + 0x3b
```

Is there no way to get WAB to report a useful exception?

**Comment by mikegoatly** [comment]
I have a user of my app reporting exactly the same problem. I've updated to 4.12.0 and the problem still persists. I'm unable to replicate the issue on any devices available to me so I suspect it is environmental, although from app insights I can very occasionally see others having the same issue.

My customer is on Windows Version: 10.0.18363.778 (x64) and has tried restarting and uninstalling/reinstalling the app to no avail.

I have logging turned on, and this is what I've received as part of the problem report:
```
2020-04-29 12:59:27:7420 Informational [MicrosoftGraphAuthenticationHelper.GetUserTokenV2Async] (False) MSAL 4.8.1.0 MSAL.UAP N/A [04/29/2020 19:59:27 - dfa2757b-1774-424c-87b4-c73e5034099b] (UnknownClient: 0.0.0.0) MSAL MSAL.UAP with assembly version '4.8.1.0'. CorrelationId(dfa2757b-1774-424c-87b4-c73e5034099b) 
2020-04-29 12:59:27:7420 Informational [MicrosoftGraphAuthenticationHelper.GetUserTokenV2Async] (False) MSAL 4.8.1.0 MSAL.UAP N/A [04/29/2020 19:59:27 - dfa2757b-1774-424c-87b4-c73e5034099b] (UnknownClient: 0.0.0.0) === InteractiveParameters Data === 
LoginHint provided: False 
User provided: False 
UseEmbeddedWebView: NotSpecified 
ExtraScopesToConsent: 
Prompt: select_account 
HasCustomWebUi: False 
 
2020-04-29 12:59:27:7420 Informational [MicrosoftGraphAuthenticationHelper.GetUserTokenV2Async] (False) MSAL 4.8.1.0 MSAL.UAP N/A [04/29/2020 19:59:27 - dfa2757b-1774-424c-87b4-c73e5034099b] (UnknownClient: 0.0.0.0) === InteractiveParameters Data === 
LoginHint provided: False 
User provided: False 
UseEmbeddedWebView: NotSpecified 
ExtraScopesToConsent: 
Prompt: select_account 
HasCustomWebUi: False 
 
2020-04-29 12:59:27:7420 Informational [MicrosoftGraphAuthenticationHelper.GetUserTokenV2Async] (False) MSAL 4.8.1.0 MSAL.UAP N/A [04/29/2020 19:59:27 - dfa2757b-1774-424c-87b4-c73e5034099b] (UnknownClient: 0.0.0.0) 
=== Request Data === 
Authority Provided? - True 
Scopes - Files.ReadWrite.All User.Read 
Extra Query Params Keys (space separated) - 
 
2020-04-29 12:59:27:7420 Informational [MicrosoftGraphAuthenticationHelper.GetUserTokenV2Async] (False) MSAL 4.8.1.0 MSAL.UAP N/A [04/29/2020 19:59:27 - dfa2757b-1774-424c-87b4-c73e5034099b] (UnknownClient: 0.0.0.0) === Token Acquisition (InteractiveRequest) started: 
  
 Authority Host: login.microsoftonline.com 
2020-04-29 12:59:27:7420 Informational [MicrosoftGraphAuthenticationHelper.GetUserTokenV2Async] (False) MSAL 4.8.1.0 MSAL.UAP N/A [04/29/2020 19:59:27 - dfa2757b-1774-424c-87b4-c73e5034099b] (UnknownClient: 0.0.0.0) Fetching instance discovery from the network from host login.microsoftonline.com 
2020-04-29 12:59:27:7576 Informational [EtwDelegateCommand.TrackExecution] Invocation completed: SignInCommand - 6ms 
2020-04-29 12:59:27:9608 Informational [MicrosoftGraphAuthenticationHelper.GetUserTokenV2Async] (False) MSAL 4.8.1.0 MSAL.UAP N/A [04/29/2020 19:59:27 - dfa2757b-1774-424c-87b4-c73e5034099b] (UnknownClient: 0.0.0.0) Resolving authority endpoints... Already resolved? - FALSE 
2020-04-29 12:59:28:4607 Warning [MicrosoftGraphAuthenticationHelper.GetUserTokenV2Async] (False) MSAL 4.8.1.0 MSAL.UAP N/A [04/29/2020 19:59:28 - dfa2757b-1774-424c-87b4-c73e5034099b] (UnknownClient: 0.0.0.0) Attempt 1 to call WAB failed 
2020-04-29 12:59:28:4607 Warning [MicrosoftGraphAuthenticationHelper.GetUserTokenV2Async] (False) MSAL 4.8.1.0 MSAL.UAP N/A [04/29/2020 19:59:28 - dfa2757b-1774-424c-87b4-c73e5034099b] (UnknownClient: 0.0.0.0) Exception type: System.Exception 
 
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21 
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x70 
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38 
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17 
   at Microsoft.Identity.Client.Platforms.uap.WebUI.<>c__DisplayClass7_0.<<InvokeWABOnMainThreadAsync>b__0>d.MoveNext() + 0x1cc 
--- End of stack trace from previous location where exception was thrown --- 
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21 
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x70 
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38 
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17 
   at Microsoft.Identity.Client.Platforms.uap.DispatcherTaskExtensions.<>c__DisplayClass0_0`1.<<RunTaskAsync>b__0>d.MoveNext() + 0x133 
--- End of stack trace from previous location where exception was thrown --- 
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21 
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x70 
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38 
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17 
   at Microsoft.Identity.Client.Platforms.uap.DispatcherTaskExtensions.<RunTaskAsync>d__0`1.MoveNext() + 0x261 
--- End of stack trace from previous location where exception was thrown --- 
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21 
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x70 
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38 
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17 
   at Microsoft.Identity.Client.Platforms.uap.WebUI.<InvokeWABOnMainThreadAsync>d__7.MoveNext() + 0x17c 
--- End of stack trace from previous location where exception was thrown --- 
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21 
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x70 
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38 
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17 
   at Microsoft.Identity.Client.Utils.RetryOperationHelper.<ExecuteWithRetryAsync>d__0`1.MoveNext() + 0x285 
 
2020-04-29 12:59:28:6482 Warning [MicrosoftGraphAuthenticationHelper.GetUserTokenV2Async] (False) MSAL 4.8.1.0 MSAL.UAP N/A [04/29/2020 19:59:28 - dfa2757b-1774-424c-87b4-c73e5034099b] (UnknownClient: 0.0.0.0) Attempt 2 to call WAB failed 
2020-04-29 12:59:28:6482 Warning [MicrosoftGraphAuthenticationHelper.GetUserTokenV2Async] (False) MSAL 4.8.1.0 MSAL.UAP N/A [04/29/2020 19:59:28 - dfa2757b-1774-424c-87b4-c73e5034099b] (UnknownClient: 0.0.0.0) Exception type: System.Exception 
 
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21 
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x70 
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38 
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17 
   at Microsoft.Identity.Client.Platforms.uap.WebUI.<>c__DisplayClass7_0.<<InvokeWABOnMainThreadAsync>b__0>d.MoveNext() + 0x1cc 
--- End of stack trace from previous location where exception was thrown --- 
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21 
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x70 
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38 
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17 
   at Microsoft.Identity.Client.Platforms.uap.DispatcherTaskExtensions.<>c__DisplayClass0_0`1.<<RunTaskAsync>b__0>d.MoveNext() + 0x133 
--- End of stack trace from previous location where exception was thrown --- 
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21 
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x70 
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38 
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17 
   at Microsoft.Identity.Client.Platforms.uap.DispatcherTaskExtensions.<RunTaskAsync>d__0`1.MoveNext() + 0x261 
--- End of stack trace from previous location where exception was thrown --- 
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21 
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x70 
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38 
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17 
   at Microsoft.Identity.Client.Platforms.uap.WebUI.<InvokeWABOnMainThreadAsync>d__7.MoveNext() + 0x17c 
--- End of stack trace from previous location where exception was thrown --- 
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21 
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x70 
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38 
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17 
   at Microsoft.Identity.Client.Utils.RetryOperationHelper.<ExecuteWithRetryAsync>d__0`1.MoveNext() + 0x285 
 
2020-04-29 12:59:28:6482 Error [MicrosoftGraphAuthenticationHelper.GetUserTokenV2Async] (False) MSAL 4.8.1.0 MSAL.UAP N/A [04/29/2020 19:59:28 - dfa2757b-1774-424c-87b4-c73e5034099b] (UnknownClient: 0.0.0.0) Exception type: System.Exception 
 
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21 
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x70 
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38 
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17 
   at Microsoft.Identity.Client.Platforms.uap.WebUI.<>c__DisplayClass7_0.<<InvokeWABOnMainThreadAsync>b__0>d.MoveNext() + 0x1cc 
--- End of stack trace from previous location where exception was thrown --- 
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21 
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x70 
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38 
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17 
   at Microsoft.Identity.Client.Platforms.uap.DispatcherTaskExtensions.<>c__DisplayClass0_0`1.<<RunTaskAsync>b__0>d.MoveNext() + 0x133 
--- End of stack trace from previous location where exception was thrown --- 
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21 
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x70 
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38 
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17 
   at Microsoft.Identity.Client.Platforms.uap.DispatcherTaskExtensions.<RunTaskAsync>d__0`1.MoveNext() + 0x261 
--- End of stack trace from previous location where exception was thrown --- 
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21 
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x70 
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38 
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17 
   at Microsoft.Identity.Client.Platforms.uap.WebUI.<InvokeWABOnMainThreadAsync>d__7.MoveNext() + 0x17c 
--- End of stack trace from previous location where exception was thrown --- 
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21 
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x70 
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38 
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17 
   at Microsoft.Identity.Client.Utils.RetryOperationHelper.<ExecuteWithRetryAsync>d__0`1.MoveNext() + 0x40b 
--- End of stack trace from previous location where exception was thrown --- 
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21 
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x70 
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38 
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17 
   at Microsoft.Identity.Client.Platforms.uap.WebUI.<AcquireAuthorizationAsync>d__6.MoveNext() + 0x211 
 
2020-04-29 12:59:28:6482 Error [MicrosoftGraphAuthenticationHelper.GetUserTokenV2Async] (False) MSAL 4.8.1.0 MSAL.UAP N/A [04/29/2020 19:59:28 - dfa2757b-1774-424c-87b4-c73e5034099b] (UnknownClient: 0.0.0.0) Exception type: Microsoft.Identity.Client.MsalClientException 
, ErrorCode: authentication_ui_failed 
---> Inner Exception Details 
Exception type: System.Exception 
 
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21 
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x70 
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38 
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17 
   at Microsoft.Identity.Client.Platforms.uap.WebUI.<>c__DisplayClass7_0.<<InvokeWABOnMainThreadAsync>b__0>d.MoveNext() + 0x1cc 
--- End of stack trace from previous location where exception was thrown --- 
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21 
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x70 
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38 
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17 
   at Microsoft.Identity.Client.Platforms.uap.DispatcherTaskExtensions.<>c__DisplayClass0_0`1.<<RunTaskAsync>b__0>d.MoveNext() + 0x133 
--- End of stack trace from previous location where exception was thrown --- 
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21 
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x70 
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38 
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17 
   at Microsoft.Identity.Client.Platforms.uap.DispatcherTaskExtensions.<RunTaskAsync>d__0`1.MoveNext() + 0x261 
--- End of stack trace from previous location where exception was thrown --- 
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21 
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x70 
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38 
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17 
   at Microsoft.Identity.Client.Platforms.uap.WebUI.<InvokeWABOnMainThreadAsync>d__7.MoveNext() + 0x17c 
--- End of stack trace from previous location where exception was thrown --- 
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21 
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x70 
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38 
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17 
   at Microsoft.Identity.Client.Utils.RetryOperationHelper.<ExecuteWithRetryAsync>d__0`1.MoveNext() + 0x40b 
--- End of stack trace from previous location where exception was thrown --- 
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21 
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x70 
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38 
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17 
   at Microsoft.Identity.Client.Platforms.uap.WebUI.<AcquireAuthorizationAsync>d__6.MoveNext() + 0x211 
 
=== End of inner exception stack trace === 
 
   at Microsoft.Identity.Client.Platforms.uap.WebUI.<AcquireAuthorizationAsync>d__6.MoveNext() + 0x2cb 
--- End of stack trace from previous location where exception was thrown --- 
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21 
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x70 
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38 
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17 
   at Microsoft.Identity.Client.Internal.Requests.InteractiveRequest.<AcquireAuthorizationAsync>d__11.MoveNext() + 0x210 
--- End of stack trace from previous location where exception was thrown --- 
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21 
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x70 
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38 
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17 
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult() + 0xb 
   at Microsoft.Identity.Client.Internal.Requests.InteractiveRequest.<ExecuteAsync>d__10.MoveNext() + 0x232 
--- End of stack trace from previous location where exception was thrown --- 
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21 
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x70 
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38 
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17 
   at Microsoft.Identity.Client.Internal.Requests.RequestBase.<RunAsync>d__14.MoveNext() + 0x308 
 
2020-04-29 12:59:28:6482 Error [RemoteDataStore.ConnectAsync] Error connecting to remote data store: Web Authentication Broker (WAB) authentication failed. To collect WAB logs, please follow https://aka.ms/msal-net-wab-logs 
```

**Comment by tipa** [comment]
I strongly believe this is a Windows problem and cannot be fixed inside this library (at least not using the Web Authentication Broker). I also had my users collect the logs as suggested in the exception and [sent them to](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1098#issuecomment-557112876) @jmprieur, unfortunately haven't heard back from him

**Comment by jmprieur** [comment]
@tipa : I'm puzzled. Unless I missed something, I don't think I received logs in my mailbox?

**Comment by bgavrilMS** [comment]
@tipa  - I'll follow up with the WAB folks - could you send me those logs? Jean-Marc didn't receive them, maybe they were blocked by spam / security filters ... bogavril at microsoft com

Also, I have a few more questions: 

a. What scopes are you asking? 
b. Does this happen for B2C local accounts or for some / all identity providers (facebook, google etc.)
b. How often / for how many users does this happen?

**Comment by kevcrooks** [comment]
I saw this older issue that suggests the error could be caused by a proxy on the machine (which may explain why we can't reproduce our end-user's behaviour), but I haven't looked in to this further: [https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1482](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1482)

**Comment by bgavrilMS** [comment]
We have not found a way to expose more details from the exception, except by going through the logs manually. 

Long term, the solution is to migrate to WAM...

**Comment by xperiandri** [comment]
@bgavrilMS what are the exact steps? Azure AD B2C auth just no longer works. That is crazy. MS auth does not work in MS Windows app.

**Comment by bgavrilMS** [comment]
@xperiandri - is this a new issue? We had repots that the UWP control that does auth started freezing. See - https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3109
